piksi_tools/console/update_view.py

```diff
@@ -54,6 +55,46 @@ def parse_version(version):
     return gitversion_parse(version)
 
 
+class NetworkCheck():
+
+    CONNECTION_TIMEOUT = 1.0
+    THREAD_INTERVAL = 5.0
+    SERVER = ("www.google.com", 80)
+
+    def __init__(self, log, cb):
+        self._cb = cb
+        self._log = log
+        self._prev_state = True
+        self._thd_trigger = Event()
+        self._check_done = Event()
+        self._worker = Thread(target=self._check_thd)
+        self._worker.daemon = True
+        self._worker.start()
+
+    def network_is_reachable(self):
+        self._check_done.clear()
+        self._thd_trigger.set()
+        self._check_done.wait()
+        return self._prev_state
+
+    def _check_thd(self):
+        while True:
+            try:
+                socket.create_connection(NetworkCheck.SERVER, NetworkCheck.CONNECTION_TIMEOUT).close()
+                if not self._prev_state:
+                    self._log("Network is reachable.")
+                self._prev_state = True
+                self._cb(True)
+            except Exception as ex:
```

Comment:
We don't need to capture `ex` if we're not using it, and we should probably make the `try/catch` just wrap the call that we expect to fail (`socket.create_connection`), since we're using the very general `Exception`.  Unless there was some other rational for this?

Reply:
lint didn't approve the bare `except:`.